ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1352 @@
+            for event_i in range(self.current_run['data'].n_stereo_events):
+                # Event and run ids
+                event_order_number = self.current_run['data'].stereo_ids[event_i][0]
+                event_id = self.current_run['data'].event_data['M1']['stereo_event_number'][event_order_number]
+                obs_id = self.current_run['number']
+
+                # Reading event data
+                event_data = self.current_run['data'].get_stereo_event_data(event_i)
+
+                # Event counter
+                data.count = counter
+
+                # Setting up the R0 container
+                data.r0.obs_id = obs_id
+                data.r0.event_id = event_id
+                data.r0.tel.clear()
+
+                # Setting up the R1 container
+                data.r1.obs_id = obs_id
+                data.r1.event_id = event_id
+                data.r1.tel.clear()
+
+                # Setting up the DL0 container
+                data.dl0.obs_id = obs_id
+                data.dl0.event_id = event_id
+                data.dl0.tel.clear()
+
+                # Filling the DL1 container with the event data
+                for tel_i, tel_id in enumerate(tels_in_file):
+                    # Creating the telescope pointing container
+                    pointing = TelescopePointingContainer()
+                    pointing.azimuth = np.deg2rad(event_data['pointing_az']) * u.rad
+                    pointing.altitude = np.deg2rad(90 - event_data['pointing_zd']) * u.rad
+                    pointing.ra = np.deg2rad(event_data['pointing_ra']) * u.rad
+                    pointing.dec = np.deg2rad(event_data['pointing_dec']) * u.rad
+
+                    # Adding the pointing container to the event data
+                    data.pointing[tel_i + 1] = pointing
+
+                    # Adding event charge and peak positions per pixel
+                    data.dl1.tel[tel_i + 1].image = event_data['{:s}_image'.format(tel_id)]
+                    data.dl1.tel[tel_i + 1].peakpos = event_data['{:s}_peak_pos'.format(tel_id)]
+                    # data.dl1.tel[i_tel + 1].badpixels = np.array(
+                    #     file['dl1/tel' + str(i_tel + 1) + '/badpixels'], dtype=np.bool)
+
+                # Adding the event arrival time
+                time_tmp = Time(event_data['mjd'], scale='utc', format='mjd')
+                data.trig.gps_time = Time(time_tmp, format='unix', scale='utc', precision=9)
+
+                # Setting the telescopes with data
+                data.r0.tels_with_data = tels_with_data
+                data.r1.tels_with_data = tels_with_data
+                data.dl0.tels_with_data = tels_with_data
+                data.trig.tels_with_trigger = tels_with_data
+
+                # Setting the instrument sub-array
+                data.inst.subarray = self.magic_subarray
+
+                yield data
+                counter += 1
+
+        return
+
+    def _mono_event_generator(self, telescope):
```

Comment:
I would propose to call this `iter_mono_events`

Reply:
OK.